### DISCUSSION TITLE
Tasmota SwitchMode Dimming Options - Shelly Dimmer 2 with Toggle Switch

### DISCUSSION THREAD

**Discussion by rmduk** [arendst/Tasmota#18436] [Q&A]
I've been using a shelly dimmer 2 with stock firmware with a simple rocker switch for on/off, and then using Home Assistant to control dimming function through Shelly intergration.  I've recently flashed this Shelly dimmer to tasmota as part of converting all my home automation controllers to Tasmota (to remove need for 3rd party cloud connections).
So running the latest tasmota firmware, and having to use switchmode 11 to get the dimmer functions available,  it means that when I turn the light on with the toggle switch it just cycles through dimming levels, as if you pushed and held a button switch.  So my question is, is it possible to make this work like I want with a rocker switch instead of a button.  From the reading I've done it looks like it might be possible to create a rule, or possibly change the template settings?  I'm hoping someone can point me in the right direction on how to do this if it is possible, or confirm it won't work so I need to revert the firmware back.  Simply changing the switch would lead to divorce so is not an option.

Many Thanks....Rob

**Comment by craigmdennis** [comment]
[You can try the following rule](https://tasmota.github.io/docs/Rules/#control-a-dimmer-with-one-switch). I am using it and it behaves as you describe.

```
Backlog SwitchMode 11; SetOption32 10; Rule1 1;

Rule1 
on system#boot do var1 + ENDON
on switch1#state=2 do POWER TOGGLE ENDON
on switch1#state=4 do DIMMER %var1% ENDON
on switch1#state=7 do event upordown=%var1% ENDON
on event#upordown=+ do var1 - ENDON
on event#upordown=- do var1 + ENDON
```

I followed https://templates.blakadder.com/shelly_Dimmer_2.html

**Reply by wewa00** [reply]
@craigmdennis it would be good to mention here, that a toggle switch needs to be connected to "SW2".

On "SW1" you can connect a push button to achieve the "control with one button" mode described here: https://tasmota.github.io/docs/Rules/#control-a-dimmer-with-one-switch

Could you or someone add this part of information to the tasmota docs?
